contentcuration/contentcuration/frontend/settings/pages/Account/StudioCopyToken.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+
+  <div class="token-input-wrapper">
+    <KCircularLoader
+      v-if="loading"
+      :shouldShow="loading"
+      size="38"
+      class="loader"
+    />
+    <template v-else>
+      <KTextbox
+        ref="tokenInput"
+        :value="displayToken"
+        readonly
+        class="notranslate token-input"
+        label=" "
+        :floatingLabel="false"
+      />
+      <KIconButton
+        icon="copy"
+        :text="$tr('copyTokenButton')"
+        class="copy-button"
+        @click="copyToClipboard"
+      />
+    </template>
+  </div>
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'StudioCopyToken',
+    props: {
+      token: { type: String, required: true },
+      loading: { type: Boolean, default: false },
+      hyphenate: { type: Boolean, default: true },
+    },
+    computed: {
+      displayToken() {
+        return this.hyphenate ? this.token.slice(0, 5) + '-' + this.token.slice(5) : this.token;
+      },
+      successMessage() {
+        return this.$tr('tokenCopied');
+      },
+    },
+    methods: {
+      copyToClipboard() {
+        if (!this.token.trim()) {
+          this.$store.dispatch('showSnackbarSimple', this.$tr('tokenCopyFailed'));
+          return;
+        }
+        if (navigator.clipboard) {
+          navigator.clipboard
+            .writeText(this.displayToken)
+            .then(() => {
+              this.$store.dispatch('showSnackbarSimple', this.successMessage);
+            })
+            .catch(() => {
+              this.$store.dispatch('showSnackbarSimple', this.$tr('tokenCopyFailed'));
+            });
+        } else {
+          this.$store.dispatch('showSnackbarSimple', this.$tr('tokenCopyFailed'));
+        }
+      },
+    },
+    $trs: {
+      copyTokenButton: 'Copy token',
+      tokenCopied: 'Token copied!',
+      tokenCopyFailed: 'Failed to copy token.',
+    },
+  };
+
+</script>
+
+
+<style scoped>
+
+  .token-input-wrapper {
+    display: flex;
+    align-items: center;
+    width: 100%;
+    height: 48px;
+    background-color: #ebebeb !important;
```

Comment:
Hardcoded colors are not to be used (see [Colors](https://design-system.learningequality.org/colors)).

Also I don't see what's the purpose of this style?